jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -295,7 +295,7 @@ public static void copy(Resource resource, Content.Sink sink, ByteBufferPool buf
             InputStream inputStream = resource.newInputStream();
             if (inputStream == null)
                 throw new IllegalArgumentException("Resource does not support InputStream: " + resource);
-            Content.Source source = Content.Source.from(new ByteBufferPool.Sized(bufferPool, direct, bufferSize), inputStream, 0, -1);
```

Comment:
see https://github.com/jetty/jetty.project/pull/12972#discussion_r2032910414 on how this restriction can be avoided